mflowgen/Tile_MemCore/construct.py

```diff
@@ -230,20 +253,40 @@ def construct():
   g.connect_by_name( drc,      debugcalibre )
   g.connect_by_name( lvs,      debugcalibre )
 
+  # Pwr aware steps:
+  if pwr_aware:
+      g.connect_by_name( power_domains,        dc           )
+      g.connect_by_name( power_domains,        init         )
+      g.connect_by_name( power_domains,        power        )
+      g.connect_by_name( power_domains,        place        )
+      g.connect_by_name( power_domains,        cts          )
+      g.connect_by_name( power_domains,        postcts_hold )
+      g.connect_by_name( power_domains,        route        )
+      g.connect_by_name( power_domains,        postroute    )
+      g.connect_by_name( power_domains,        signoff      )
+      #g.connect(power_domains.o('pd-globalnetconnect.tcl'), power.i('globalnetconnect.tcl'))
+
   #-----------------------------------------------------------------------
   # Parameterize
   #-----------------------------------------------------------------------
 
   g.update_params( parameters )
 
+  # Update PWR_AWARE variable
+  if pwr_aware:
+      dc.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, True )
+      init.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, True )
+      power.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, True )
+
+
   # Disable pwr aware flow
-  init.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, allow_new=True )
-  power.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, allow_new=True  )
+  #init.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, allow_new=True )
+  #power.update_params( { 'PWR_AWARE': parameters['PWR_AWARE'] }, allow_new=True  )
```

Comment:
This code should be executed unconditionally. Even if pwr_aware is false, the parameter still needs to be updated because the scripts reference it. I see that the same bug is present in the pe tile construct.py, so it would be good to fix this in both places.